### ISSUE TITLE
Using source map doubles memory usage

### ISSUE THREAD

**Issue by ninja-** [vercel/ncc#596]
Running app after compiling with -s option uses ~ 320MB on start, while without the flag it's 150-200MB.
Running pure typescript compiled version, even with require('source-map-support'), does not have this problem

**Comment by ninja-** [comment]
I did a test, replaced the custom source map require in index.js with regular require('source-map-support')
It makes the map lazy load at least, but on the first Error printed memory usage jumps from 120MB to 360MB :/

**Comment by ninja-** [comment]
Using 

`nosources-source-map` as devtool seems like a much better choice, it provides line numbers and original files but without using all the source code that you're not going to use anyway without chrome debugger.
It doesn't seem to help that much with mem usage though...but map is now 30% previous size

**Comment by ninja-** [comment]
I tried to use nodejs native integration --enable-source-maps instead, and it seems to work better.
But still, when it tries to read the sourcemap there is big memory usage jump - but at least it later garbage collects the map...

**Comment by styfle** [comment]
Now that [`--enable-source-maps`](https://nodejs.org/dist/latest-v16.x/docs/api/cli.html#cli_enable_source_maps) is stable and supported back to Node.js 12, we could probably avoid the patching done to support older versions of node and expect the user to run `node --enable-source-maps dist/index.js` if they want source map support at runtime.

**Comment by ninja-** [comment]
yes native source map seems fine, but the key is to split index.js into chunks otherwise the sourcemap would get super big and take a ton of memory when loaded

**Comment by styfle** [comment]
The purpose of ncc is to bundle chunks of js into a single file. So splitting doesn't make sense, you could just avoid using ncc and keep the original files.

**Comment by ninja-** [comment]
I disagree with that, having index.js automatically split into 10 chunks and required in main index.js would serve the same purpose as currently but would fix the issues with source maps
